src/lib/core/sio.c

```diff
@@ -55,38 +55,39 @@ static_assert(SMALL_STATIC_SIZE > NI_MAXHOST + NI_MAXSERV,
  * checks and all.
  */
 static int
-sio_socketname_to_buffer(int fd, char *buf, int size)
+sio_socketname_to_buffer(int fd, const struct sockaddr *base_addr,
+			 socklen_t addrlen, char *buf, int size)
 {
 	int n = 0;
 	(void)n;
 	SNPRINT(n, snprintf, buf, size, "fd %d", fd);
 	if (fd < 0)
 		return 0;
-	struct sockaddr_storage addr;
-	socklen_t addrlen = sizeof(addr);
-	struct sockaddr *base_addr = (struct sockaddr *)&addr;
-	int rc = getsockname(fd, base_addr, &addrlen);
-	if (rc == 0) {
+	if (base_addr != NULL) {
 		SNPRINT(n, snprintf, buf, size, ", aka ");
 		SNPRINT(n, sio_addr_snprintf, buf, size, base_addr, addrlen);
 	}
-	addrlen = sizeof(addr);
-	rc = getpeername(fd, (struct sockaddr *) &addr, &addrlen);
+	struct sockaddr_storage peer_addr_storage;
```

Comment:
It seems to me, that this peer searching should not belong to this function. IMHO, `getpeername` should be moved out to the `sio_socketname_addr` so that `sio_socketname_to_buffer` would only dump info it gets and wouldn't try to find out anything about `fd`.

Reply:
Fixed